### PR TITLE
Add unique index to contribution_info_publishers

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/7838

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Enable Rewards
- Claim promotion
- Make a contribution
- Ensure that contribution is deducted
- Ensure that data is correct in database (`contribution_info_publishers` table)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by NejcZdovc** [comment]
CI failed on windows, restarting

**Comment by NejcZdovc** [comment]
could you please add browser test for upgrade case so that we make sure that no data was deleted from contribution_info_publishers table

**Comment by NejcZdovc** [comment]
please add test plan

File: vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc [hunk 1]
```diff
@@ -96,6 +99,13 @@ bool DatabaseContributionQueue::MigrateToV15(
   return publishers_->Migrate(transaction, 15);
 }
 
+bool DatabaseContributionQueue::MigrateToV21(
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc:102]: This is different table than ContributionInfo. I think this is not needed

Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/database/database_contribution_queue.cc]: OK, I see. Fixed now.

**Comment by NejcZdovc** [comment]
CI failed on lint https://ci.brave.com/job/pr-brave-browser-rewards-unique-index-contribution-info-publishers/12/execution/node/290/log/

**Comment by emerick** [comment]
Fixed linter issue and kicked off CI.

**Comment by NejcZdovc** [comment]
CI was aborted on windows, restarting

**Pull request merged**